scripts/ci/images/ci_start_arm_instance_and_connect_to_docker.sh

```diff
@@ -20,7 +20,7 @@ SCRIPTS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
 WORKING_DIR="/tmp/armdocker"
 INSTANCE_INFO="${WORKING_DIR}/instance_info.json"
 ARM_AMI="ami-0e43196369d299715"  # AMI ID of latest arm-docker-ami-v*
```

Comment:
Not related to this PR, but better get AMI from the SSM in this case we could be sure that we use latest AMI with security patches

Reply:
I guess build under the QEMU in one go is not our path?

Reply:
Qemu build takes about 2 hours to complete :(

Reply:
Or I think close to 3 hours if all is well. Generally the rate is 16x slower. so as long the price is not 16x cheaper we are at the same cost but way smaller "elapsed" time with ARM

Reply:
I'm not sure maybe this numbers a bit outdated.

In my case it took 48 minutes to build prod image on ARM, and it took at least 10 minutes to build `gevent` which do not have pre-built wheels for ARM  on Python 3.8

```console
❯ docker build --platform linux/arm64 . \
    --build-arg DOCKER_CONTEXT_FILES=./docker-context-files \
    --build-arg INSTALL_MYSQL_CLIENT="true" \
    --build-arg INSTALL_MSSQL_CLIENT="false" \
    --build-arg INSTALL_POSTGRES_CLIENT="true" \
    --no-cache \
    --tag "airflow-image:non-native-arm64"
[+] Building 2915.0s (68/68) FINISHED                                                                                                          docker:default
```

Which slower for about 4 than native X86_64 build, which took 12 minutes
```console
❯ docker build --platform linux/amd64 . \
    --build-arg DOCKER_CONTEXT_FILES=./docker-context-files \
    --build-arg INSTALL_MYSQL_CLIENT="true" \
    --build-arg INSTALL_MSSQL_CLIENT="false" \
    --build-arg INSTALL_POSTGRES_CLIENT="true" \
    --no-cache \
    --tag "airflow-image:native-x86_64"
[+] Building 778.8s (70/70) FINISHED                                                                                                           docker:default
```

Reply:
The 16x times I last saw (you can was mostly for CI image builds - which have much higher proportion of what needed to be compiled comparing to a number of downloads and it was done at the time when there were many more packages that needed to be compiled on top - many of them, when we first switched to ARM. I think that 16x was what I saw at that time as the raw "emulator" overhead in some of those

It's likely heavily outdated and maybe they were in some specific circumstances. Last time I looked at it was like 1.5 years ago. So yyeah 4x/5x is likely good estimate for PROD images. 

But also the ARM instances are not 10x more expensive. This is just the max price we are willing to pay.  Usually the spot price is way lower than that - last time I checked the AMD/ARM prices were comparable (certainly not 10x) and we are talking about total $15(!) USD / month for all the time of ARM instances running and that.  So saving few USD is proably not worth even thinking about it :D.

The good thing about using ARM instances for that is that we can build the multi-platform images easily by running the two in parallel https://github.com/apache/airflow/actions/runs/7739332335/job/21102024869#step:11:7730 - building both ARM and AMD image using buildx multi-builder capabilities takes 5-8 minutes, depends on Python version. And in this case the "elapsed time" trumps the cost (but I still think that even if it is 4x longer it's still far cheaper to use the ARM spot instances for that). 

But -- of course - maybe you want to look at the setup and try to optimize it :).  

In the future we could (and this is something if we would like to use public ARM instance) much complicate the build process and build images for AMD and ARM in separate jobs and combine them as well.